src/main/scala/jtag/JtagStateMachine.scala

```diff
@@ -76,13 +76,10 @@ class JtagStateMachine(implicit val p: Parameters) extends Module() {
   }
   val io = IO(new StateMachineIO)
 
-  val nextState = WireInit(JtagState.TestLogicReset.U)
-  val currStateReg = Module (new AsyncResetRegVec(w = JtagState.State.width,
-    init = JtagState.State.toInt(JtagState.TestLogicReset)))
-  currStateReg.io.en := true.B
-  currStateReg.io.d  := nextState
-  val currState = currStateReg.io.q
+  val nextState = Wire(UInt(JtagState.State.width.W))
```

Comment:
why did you get rid of the `WireInit` here in favor of the two-line version for `nextState`?

Reply:
Fixed